packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -94,6 +123,10 @@ class NavigatorState extends State<Navigator> {
 
   void dispose() {
     config.observer?._navigator = null;
+    for (Route route in _history) {
```

Comment:
yeah, I can use the _debugLocked mechanism to make the stack traces better. If anyone does try to mutate the list, it'll crash when they get back here, the way the code is written now.